### PR TITLE
Inject iproxy into context

### LINKED ISSUE #61750: Inject iproxy object in context to support different versions
Internal tooling still on an older version of iproxy, which is conflicting with the new libusbmuxd 2.0.2 port parameter format:
https://github.com/flutter/flutter/pull/59512/files#diff-4e6a9727ba10772d640b597dbdd61d03R835-R836

Inject an `IProxy` object into the context to support different versions parameter formats.

Tracking b/160845834#comment6

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Create new `IProxy` object to wrap call to `iproxy` to support g3 lagging versions.  This caused issues when the new version of `iproxy` [changed its parameters](https://github.com/flutter/flutter/pull/59512/files#diff-4e6a9727ba10772d640b597dbdd61d03R835-R836) from `iproxy device-port host-port device-id` to `proxy device-port:host-port --udid device-id`.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/61750
https://github.com/flutter/flutter/pull/59512

## Tests

Updated tests

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/iproxy.dart
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/ios/iproxy.dart:14]: nit: small doc comment describing what it does

File: packages/flutter_tools/lib/src/ios/iproxy.dart
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/ios/iproxy.dart:46]: doesn't need to be `async`

**Review by jonahwilliams — APPROVED** [review]
LGTM

Seems like a nice improvement anyway

**Comment by jmagman** [comment]
I kicked off FRoB manually so I didn't need to wait for approval, but the diff didn't go cleanly since there are pending changes between what's in g3 and this PR, close to the same line changes, and FRoB is conservative about these conflicts.  I cloned the CL and manually added the missing lines, and shockingly there were no build issues changing the required parameters.
cl/321857203

File: packages/flutter_tools/lib/src/context_runner.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/context_runner.dart:177]: @jonahwilliams another option is to remove this from the high level context, and instead make it required to initialize `XCDevice` since that's the only place it's used.
https://github.com/flutter/flutter/blob/859fce9010895ce5bcdb913a8373b5b1a194a1fb/packages/flutter_tools/lib/src/context_runner.dart#L250-L257

would become:
```dart
 XCDevice: () => XCDevice( 
   processManager: globals.processManager, 
   logger: globals.logger, 
   artifacts: globals.artifacts, 
   cache: globals.cache, 
   platform: globals.platform, 
   xcode: globals.xcode, 
   iproxy: IProxy(
     iproxyPath: globals.artifacts.getArtifactPath(
       Artifact.iproxy,
       platform: TargetPlatform.ios,
     ),
     logger: globals.logger,
     processManager: globals.processManager,
     dyLdLibEntry: globals.cache.dyLdLibEntry,
   ), 
```

This would make
https://github.com/flutter/flutter/pull/61762/files#diff-ebc648b2067d73f2af30539f5d3a59faR503
become:
```dart
iProxy: _iproxy,
```

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/context_runner.dart:177]: IMO Leaving/Removing from the context injection isn't really that important. As long as we're not introducing more globals into classes that have already been cleaned up, I have a separate plan to remove the context w/o breaking g3

Comment (jmagman) [reply packages/flutter_tools/lib/src/context_runner.dart:177]: `XCDevice` is close to being cleaned up, I already did this exercise with https://github.com/flutter/flutter/pull/53203 (which was further pushed down in https://github.com/flutter/flutter/pull/5330,  we can't do in this case since g3 wants to override). I convinced myself, I'm going to do this pattern from the beginning.

**Review by jonahwilliams — APPROVED** [review]
Still LGTM

**Pull request merged**
